data/trie/sync.go

```diff
@@ -186,6 +176,22 @@ func (ts *trieSyncer) checkIfSynced() (bool, error) {
 			newElement = newElement || tmpNewElement
 
 			delete(ts.nodesForTrie, nodeHash)
+
+			err = encodeNodeAndCommitToDB(currentNode, ts.trie.Database())
+			if err != nil {
+				return false, err
+			}
+
+			if !ts.rootFound && bytes.Equal([]byte(nodeHash), ts.rootHash) {
+				var collapsedRoot node
+				collapsedRoot, err = currentNode.getCollapsed()
+				if err != nil {
+					return false, err
+				}
+
+				ts.trie.root = collapsedRoot
+				ts.rootFound = true
```

Comment:
so we set root only after finding all its childs ?

Reply:
Yes, it should have no impact upon the functioning of the algorithm.